### ISSUE TITLE
Unable To Publish To Github Package Registry !!

### ISSUE THREAD

**Issue by varunsridharan** [JS-DevTools/npm-publish#14]
I have tried to use this action to publish in Github Package Registry .

but it keeps failing 

here is the action log
https://github.com/varunsridharan/npm-proxy/runs/1177247171?check_suite_focus=true

Note i did merge #12 in my fork and did a small change but still can't get it to work.

Changes I did : https://github.com/g-actions/npm-publish/commit/5a1378e5e0ff3500e1374298be30dd409a5bfe37

**Comment by JamesMessinger** [comment]
@varunsridharan - I can't access the GitHub repo that you linked to, so I can't view the log.  Can you post it here instead?  Also, can you be sure to [enable debug logging in GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/enabling-debug-logging) so it includes more details, please?

**Comment by varunsridharan** [comment]
@JamesMessinger sorry about that. well i was working on trying to things working. so deleted the repo and keep working 
And finally got it work just making a small change even after merging #12 

please check this commit : g-actions@5a1378e

**Comment by varunsridharan** [comment]
@JamesMessinger and also i found that we can't publish ***scoped*** npm packages using this action for that i changed 

and i was able to get things working with scoped package by making below changes

```diff
- let command = options.dryRun ? [ "publish","--dry-run" ] : ["publish" ];
+ let command = options.dryRun ? [ "publish","--dry-run","--access","public" ] : ["publish","--access","public" ];
```

**Comment by JamesMessinger** [comment]
@varunsridharan - The bug with it not publishing a new package should be fixed now.  I've merged PR #12 which effectively does the same thing as your workaround.

The action _does_ work with scoped NPM packages, but NPM defaults to making scoped packages private, which requires a paid account.  A workaround would be to add an `access` option to the GitHub Action, which would allow overriding the default package access (using the `--access` argument).

**Comment by varunsridharan** [comment]
well even after merging #12 pull request it did not work when new package is published so i had to make a small change 
please check this commit : g-actions/npm-publish@5a1378e

**Comment by JamesMessinger** [comment]
It looks like you modified the `dist` folder directly, rather than re-building from `src`.

**Comment by varunsridharan** [comment]
yes i did !!

**Comment by JamesMessinger** [comment]
@varunsridharan - I just published version 1.4.0, which adds an `access` input parameter (which gets passed to NPM's `--access` argument).  This should satisfy your needs and resolve all the issues you mentioned above.

**Comment by ZitRos** [comment]
@JamesMessinger @varunsridharan 

Hello! The author of #12 here. Turns out, not only you can make a wrong PR but also a wrong test for a wrong PR that all seem valid :D But the issue is quite strange itself: I spend around an hour of debugging in a dance to figure out what's wrong.

Let's start with #12 - [this](https://github.com/JS-DevTools/npm-publish/pull/12/commits/2656c73ee8b0b84229184d0548dc0373ddfda812#diff-7ed38395221a42cf5b3ccf6754c4ef75e4713032fd607cbac04ce51849597466R30-R36) was incorrect, because NPM returns the exit status code `1`, and turns out `ezSpawn.async` throws in this case.

When I understood it, I fixed it to return the exit code `1` in a test and moved [this](https://github.com/JS-DevTools/npm-publish/pull/12/commits/2656c73ee8b0b84229184d0548dc0373ddfda812#diff-7ed38395221a42cf5b3ccf6754c4ef75e4713032fd607cbac04ce51849597466R33-R36) to a catch clause. However, perhaps of the way how `npm.mock` works, this didn't pass other tests (f.e. "throw when NPM throws an error")!

So I found the only working magic combination of the code to satisfy both the test and a real state of things which I am contributing now by #19.
